tests/integration/build-operator.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+
+# Copyright 2022 Red Hat, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Exit immediately for non zero status
+set -e
+# Check unset variables
+set -u
+# Print commands
+set -x
+
+WD=$(dirname "$0")
+WD=$(cd "$WD"; pwd)
+
+# build and push operator image
+build_operator_image() { # compile operator source and push operator image
+    local TAG
+    TAG="$(git rev-parse HEAD)"
+    local BUILDER_CLI="${BUILDER_CLI:-docker}"
+    # var name IMAGE is in kind provisioner
+    export OPERATOR_IMAGE="quay.io/maistra-dev/istio-operator-integ:${TAG}"
```

Comment:
You can use the kind registry for that, no need to use quay or any external registry.
Take a look at how Istio integration tests build the images, for example

Reply:
Think of a regular, anonymous user. They should be able to run all the build & tests without having permission to push to quay.